Zend/zend_exceptions.c

```diff
@@ -361,15 +361,18 @@ ZEND_METHOD(ErrorException, __construct)
 	ZVAL_LONG(&tmp, severity);
 	zend_update_property_ex(zend_ce_exception, Z_OBJ_P(object), ZSTR_KNOWN(ZEND_STR_SEVERITY), &tmp);
 
-	if (argc >= 4) {
+	if (filename) {
 		ZVAL_STR_COPY(&tmp, filename);
 		zend_update_property_ex(zend_ce_exception, Z_OBJ_P(object), ZSTR_KNOWN(ZEND_STR_FILE), &tmp);
 		zval_ptr_dtor(&tmp);
-    	if (argc < 5) {
-    	    lineno = 0; /* invalidate lineno */
-    	}
+	}
+
+	if (!lineno_is_null) {
```

Comment:
Is it worth to add a validation that `lineno` can't be negative?

Reply:
Should probably be an assertion no? As userland can't set it.

Reply:
I think this class behaves just like any other classes, so it's instantiable in userland (e.g. in my test). But most probably this isn't worth the effort.

Reply:
I'm pretty sure I have code somewhere using -1 dummy line numbers :P